### DISCUSSION TITLE
USB in Hyper V

### DISCUSSION THREAD

**Discussion by decipher3114** [dorssel/usbipd-win#267] [Q&A] [answered]
How can I use my wifi adapter in Hyper-V??
Anyone Please tell...

**Comment by dorssel** [comment, marked as answer]
1) Install a USBIP server on the Windows host (e.g. `usbipd-win`). Share the device as documented.
2) Install a USBIP client in the VM and follow its setup instuctions.

**Reply by decipher3114** [reply]
> 1. Install a USBIP server on the Windows host (e.g. `usbipd-win`). Share the device as documented.
> 2. Install a USBIP client in the VM and follow its setup instuctions.

Sir what command i need to enter in linux inside VM.
as I read that usbip --remote=<HOST> -b 1-1

so what i need to enter in place of host.

**Reply by decipher3114** [reply]
![image](https://user-images.githubusercontent.com/84389932/153995374-fde1b42a-22e9-4283-86ba-ac32601e1ae7.png)
 I got this error. So, could you please tell what could be problem.

Sorry ,the image is not full.

**Comment by decipher3114** [comment]
![image](https://user-images.githubusercontent.com/84389932/153995775-638ddae2-5a25-4e4a-90e9-486d6b60d8b5.png)
This is the full error

**Comment by decipher3114** [comment]
![image](https://user-images.githubusercontent.com/84389932/154005344-12d31a7e-56f6-4cbd-b412-85e979137388.png)
I corrected the error now but it now shows me to import it.
But there is no option to import it.

**Reply by dorssel** [reply]
You are nearly there. The command `usbip attach -r 192.168.56.1 -b 1-1` is correct. Only problem is: it fails. Now it is time to figure out why. Normally, the usbip client gives a little more info. May I ask what OS this is?

To debug, you can run the server-side with debugging. See https://github.com/dorssel/usbipd-win/wiki/Troubleshooting. That will hopefully tell you why the attach fails. Did you get any warnings when you did the `bind` server side?

**Reply by decipher3114** [reply]
> You are nearly there. The command `usbip attach -r 192.168.56.1 -b 1-1` is correct. Only problem is: it fails. Now it is time to figure out why. Normally, the usbip client gives a little more info. May I ask what OS this is?
> 
> To debug, you can run the server-side with debugging. See https://github.com/dorssel/usbipd-win/wiki/Troubleshooting. That will hopefully tell you why the attach fails. Did you get any warnings when you did the `bind` server side?

No error while binding on Windows.
It is Kali Linux
If the problem is on Windows Side, it would have shown the error.
But it don't show any.

Can it be bcz of Enhanced Session on Kali
Since it runs as RDP. So, can this be a problem??

Or the windows need any driver or other things
Bcz Linux has fully upgraded to 2022.1

So, could you please try it on your side.

**Reply by dorssel** [reply]
I use usbipd on Hyper-V all the time. But never with Kali. You will have to troubleshoot that yourself. Follow the instructions I pointed out earlier and get the server logs.

**Reply by decipher3114** [reply]
> I use usbipd on Hyper-V all the time. But never with Kali. You will have to troubleshoot that yourself. Follow the instructions I pointed out earlier and get the server logs.

![image](https://user-images.githubusercontent.com/84389932/154054481-a4dcd103-15ce-47a3-b070-09865c173f6b.png)

**Reply by decipher3114** [reply]
> I use usbipd on Hyper-V all the time. But never with Kali. You will have to troubleshoot that yourself. Follow the instructions I pointed out earlier and get the server logs.

Sorry Sir, but the biggest mistake i was doing was Sudo
LOL and LOL
Laugh Out Loud and Lots of Love

You are great sir.

**Reply by dorssel** [reply]
It is supposed to look like this:
```
PS C:\> sc stop usbipd

SERVICE_NAME: usbipd
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 3  STOP_PENDING
                                (STOPPABLE, NOT_PAUSABLE, ACCEPTS_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
PS C:\> sc query usbipd

SERVICE_NAME: usbipd
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
PS C:\>
```

**Comment by rmshub** [comment]
Hi @decipher3114,
I got similar error, "open vhci_driver". How it is resolved ?
Could you please share procedure you followed ?

**Reply by decipher3114** [reply]
There is a video I made about it on YouTube
https://youtu.be/ztXFGX0WcJU
